packages/langium/src/grammar/lsp/grammar-completion-provider.ts

```diff
@@ -0,0 +1,93 @@
+
+import type { Range } from 'vscode-languageserver-types';
+import { CompletionItemKind } from 'vscode-languageserver-types';
+import type { NextFeature } from '../../lsp/completion/follow-element-computation';
+import type { CompletionAcceptor, CompletionContext } from '../../lsp/completion/completion-provider';
+import { DefaultCompletionProvider } from '../../lsp/completion/completion-provider';
+import type { LangiumServices } from '../../services';
+import type { MaybePromise } from '../../utils/promise-util';
+import { getContainerOfType } from '../../utils/ast-util';
+import { equalURI, relativeURI } from '../../utils/uri-util';
+import type { LangiumDocument, LangiumDocuments } from '../../workspace';
+import type { AbstractElement } from '../generated/ast';
+import { isAssignment } from '../generated/ast';
+import { Utils } from 'vscode-uri';
+
+export class LangiumGrammarCompletionProvider extends DefaultCompletionProvider {
+
+    private readonly documents: () => LangiumDocuments;
+
+    constructor(services: LangiumServices) {
+        super(services);
+        this.documents = () => services.shared.workspace.LangiumDocuments;
+    }
+
+    protected override completionFor(context: CompletionContext, next: NextFeature<AbstractElement>, acceptor: CompletionAcceptor): MaybePromise<void> {
+        const feature = getContainerOfType(next.feature, isAssignment);
+        if (feature?.feature === 'path') {
+            this.completeImportPath(context, acceptor);
+        } else {
+            return super.completionFor(context, next, acceptor);
+        }
+    }
+
+    private completeImportPath(context: CompletionContext, acceptor: CompletionAcceptor): void {
+        const text = context.textDocument.getText();
+        const existingText = text.substring(context.tokenOffset, context.offset);
+        let allPaths = this.getAllFiles(context.document);
+        let range: Range = {
+            start: context.position,
+            end: context.position
+        };
+        if (existingText.length > 0) {
+            const existingPath = existingText.substring(1);
+            allPaths = allPaths.filter(path => path.startsWith(existingPath));
+            // Completely replace the current token
+            const start = context.textDocument.positionAt(context.tokenOffset + 1);
+            const end = context.textDocument.positionAt(context.tokenEndOffset - 1);
+            range = {
+                start,
+                end
+            };
+        }
+        for (const path of allPaths) {
+            // Only insert quotes if there is no `path` token yet.
+            const delimiter = existingText.length > 0 ? '' : '"';
+            const completionValue = `${delimiter}${path}${delimiter}`;
+            acceptor(context, {
+                label: path,
+                textEdit: {
+                    newText: completionValue,
+                    range
+                },
+                kind: CompletionItemKind.File,
```

Comment:
In case we have already a path: `import <|>"../common"`  one could remove `"../common"` using a textEdit

Reply:
I don't think we should do that:

1. The completion context is generally unaware of any token that starts at or after the cursor position. This is the expected behavior for completion, as it should be independent of any following tokens.
2. It works the same way in TypeScript, where text is replaced if you're inside of a string constant, whereas text is inserted if you're at the start of a string constant.
![image](https://github.com/langium/langium/assets/4377073/e852ccc7-9ae0-4b99-878a-5a8807ac4090)
![image](https://github.com/langium/langium/assets/4377073/80cb33d7-81cd-4bd6-9b98-04f2b3fa7f39)